### ISSUE TITLE
Some methods in ILCursor cannot be used by code mods

### ISSUE THREAD

**Issue by max4805** [EverestAPI/Everest#45]
I have been developing a code mod using `IL.Celeste` recently, and I ran into some issues when trying to use `ILCursor`. I cannot use some of the methods without the game blowing up with various errors, for example:
* `TryGotoNext(MoveType moveType, params Func<Instruction, bool>[] predicates)`: "The JIT compiler encountered an internal limitation"
```
Ver 1.2.6.1 [Everest: 859-azure-98148]
07/17/2019 13:00:35
System.InvalidProgramException: Le compilateur JIT a rencontré une limitation interne.
```
* `GotoNext(MoveType moveType, params Func<Instruction, bool>[] predicates)`:
```
Ver 1.2.6.1 [Everest: 859-azure-98148]
07/17/2019 13:01:51
System.NullReferenceException: Object reference not set to an instance of an object.
   at Celeste.Mod.Helpers.MonoModUpdateShim._ILCursor.GotoNext(ILCursor c, Func`2[] predicates) in d:\a\1\s\Celeste.Mod.mm\Mod\Helpers\MonoModUpdateShim.cs:line 31
```
* `TryGotoNext(params Func<Instruction, bool>[] predicates)`: works fine

I am opening this issue because from what I understand, these crashes are coming from `ILCursor` getting relinked to `MonoModUpdateShim`. If I remove that shim my code mod runs fine (but I guess it breaks older code mods).

I would love to help solving that issue, but I am afraid of breaking backward compatibility. Which mods should I test my changes with if I try to solve this?

**Comment by 0x0ade** [comment]
Your guess is right, `Everest.Relinker` adds a few rules to relink mods using older methods in MonoMod to `MonoModUpdateShim`. [EverestTAS-CelesteInterop](https://github.com/EverestAPI/CelesteTAS-EverestInterop) is a good candidate to check backwards compatibility with. I'll see if I can fix that before heading out for my vacation, but you're welcome to fix that as well :)

**Comment by max4805** [comment]
What I ended up doing is cloning CelesteTAS-EverestInterop and adding some pointless calls to HookILCursor in it (to ensure all variants of Goto* and Find* work) and trying to make it work along with my code mod.

... and it seems like some methods in `MonoModUpdateShim` are not necessary (honestly, I don't have any idea why, but everything seems to work without them, even if they are called 🤷‍♂). I also added some `[ShimFromAttribute]`s to prevent unwanted relinking. Here is what I did: https://github.com/max4805/Everest/commit/08375547877b44354fb4a52893285379f13bb4c0
